server/src/main/java/io/crate/planner/operators/Eval.java

```diff
@@ -108,14 +112,20 @@ public FetchRewrite rewriteToFetch(TableStats tableStats, Collection<Symbol> use
         if (fetchRewrite == null) {
             return null;
         }
+        LogicalPlan newSource = fetchRewrite.newPlan();
         Function<Symbol, Symbol> mapToFetchStubs = fetchRewrite.mapToFetchStubs();
         LinkedHashMap<Symbol, Symbol> newReplacedOutputs = new LinkedHashMap<>();
+        ArrayList<Symbol> newOutputs = new ArrayList<>();
+        for (Symbol sourceOutput : newSource.outputs()) {
+            if (sourceOutput instanceof FetchMarker) {
+                newOutputs.add(sourceOutput);
+            }
+        }
         for (Symbol output : outputs) {
             newReplacedOutputs.put(output, mapToFetchStubs.apply(output));
+            SymbolVisitors.intersection(output, newSource.outputs(), newOutputs::add);
         }
-        // Skip the Eval operator,
-        // the evaluations that the `Eval` operator took care of are now part of the replacedOutputs.
```

Comment:
If the `Eval` operator dropped columns we still need it, otherwise a parent `Rename` may fail because it expects the same amount of outputs.